### PR TITLE
Remove cadl azure core models in generated code

### PR DESCRIPTION
**Pull request by qiaozha**
fixes https://github.com/Azure/autorest.typescript/issues/1452

### CODE REVIEW

File: .scripts/ci.yml [hunk 1]
```diff
@@ -100,20 +100,16 @@ stages:
           matrix:
             macOS_Node16:
               Pool:
-              OSVmImage: "macOS-12"
-              NodeTestVersion: "16.x"
-            Browser_Linux_Node16:
-              Pool: ${{ parameters.LinuxPool }}
-              OSVmImage:
-              NodeTestVersion: "16.x"
-            Linux_Node14:
```
Comment (MaryGao) [inline .scripts/ci.yml:109]: Why we remove the above two envs?

Comment (qiaozha) [reply .scripts/ci.yml:109]: they are duplicated

File: packages/cadl-typescript/src/modelUtils.ts [hunk 2]
```diff
@@ -289,6 +293,35 @@ function isSchemaProperty(program: Program, property: ModelProperty) {
   return !(headerInfo || queryInfo || pathInfo || statusCodeinfo);
 }
 
+function extractPagedMetadataNested(
+  program: Program,
```
Comment (MaryGao) [inline packages/cadl-typescript/src/modelUtils.ts:297]: I don't think we need to define this func again.
https://github.com/Azure/autorest.typescript/blob/3ee5d2733b8be991c7787937a6fd4ad554f1a3ea/packages/cadl-typescript/src/operationUtil.ts#L160

File: packages/rlc-common/src/buildObjectTypes.ts [hunk 3]
```diff
@@ -46,6 +49,40 @@ export function buildObjectInterfaces(
   return objectInterfaces;
 }
 
+export function buildObjectAliases(
```
Comment (MaryGao) [inline packages/rlc-common/src/buildObjectTypes.ts:52]:
```suggestion
export function buildPagedObjectAliases(
```

File: packages/rlc-common/src/buildResponseTypes.ts [hunk 4]
```diff
@@ -20,10 +20,12 @@ import {
   getResponseTypeName
 } from "./helpers/nameConstructors.js";
 
+let hasErrorResponse = false;
```
Comment (MaryGao) [inline packages/rlc-common/src/buildResponseTypes.ts:23]: Not sure there will be a bug in multi-client package.

When we have two clients but only one of them imports error response, but the other will not.

Comment (qiaozha) [reply packages/rlc-common/src/buildResponseTypes.ts:23]: already initate the value as false at line 28. each client will get a false at first.

File: packages/rlc-common/src/buildObjectTypes.ts (hunk 3 above)
Comment (MaryGao) [inline packages/rlc-common/src/buildObjectTypes.ts:52]: Do we still need to build aliases expect the paged cases?

Comment (qiaozha) [reply packages/rlc-common/src/buildObjectTypes.ts:52]: Not sure I get your point, Paged has already been added into the core-paging.

File: packages/cadl-rlc-test/test/contoso/cadl-output/review/contosowidgetmanager-rest.api.md [hunk 5]
```diff
@@ -7,8 +7,11 @@
 import { Client } from '@azure-rest/core-client';
 import { ClientOptions } from '@azure-rest/core-client';
 import { CreateHttpPollerOptions } from '@azure/core-lro';
+import { ErrorModel } from '@azure-rest/core-client';
+import { ErrorResponse } from '@azure-rest/core-client';
 import { HttpResponse } from '@azure-rest/core-client';
 import { OperationState } from '@azure/core-lro';
+import { Paged } from '@azure/core-paging';
```
Comment (qiaozha) [inline packages/cadl-rlc-test/test/contoso/cadl-output/review/contosowidgetmanager-rest.api.md:14]: see examples of Paged here

File: packages/rlc-common/src/interfaces.ts [hunk 6]
```diff
@@ -141,6 +141,9 @@ export interface Schema {
   default?: any;
   readOnly?: boolean;
   usage?: SchemaContext[];
+  alias?: string;
+  outputAlias?: string;
```
Comment (MaryGao) [inline packages/rlc-common/src/interfaces.ts:145]: I feel like our alias logic is sperated into different places, sometimes in schema, sometimes in responses.

I smelled that we need to refactor the alias logic if the modelUtil could have it, it's better to move necessary logic into there.

File: packages/rlc-common/src/buildObjectTypes.ts (hunk 3 above)
Comment (MaryGao) [reply packages/rlc-common/src/buildObjectTypes.ts:52]: I mean the `buildObjectAliases` is to get all alias definition.

Currently we have one case to generate alias, that is paged.

Comment (qiaozha) [reply packages/rlc-common/src/buildObjectTypes.ts:52]: no, for polymorphism, we also use alias.

File: packages/rlc-common/src/interfaces.ts (hunk 6 above)
Comment (qiaozha) [reply packages/rlc-common/src/interfaces.ts:145]: when do we use alias in response ?

Comment (MaryGao) [reply packages/rlc-common/src/interfaces.ts:145]: Maybe I remember wrongly but we have alias model in merge+path, right?

**Review by MaryGao — COMMENTED** [review]
I just left small comments.

@joheredi Could you help review this pr?

File: packages/rlc-common/src/interfaces.ts (hunk 6 above)
Comment (qiaozha) [reply packages/rlc-common/src/interfaces.ts:145]: sync offline, would like keep merge and patch as it is.

**Review by joheredi — APPROVED** [review]

**Pull request merged**